components/containers/core/browser/settings_page_handler.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/containers/core/browser/settings_page_handler.h"
+
+#include <utility>
+
+#include "base/uuid.h"
+#include "brave/components/containers/core/browser/pref_names.h"
+#include "brave/components/containers/core/browser/prefs.h"
+
+namespace containers {
+
+SettingsPageHandler::SettingsPageHandler(
+    mojo::PendingRemote<mojom::SettingsPage> page,
+    PrefService* prefs,
+    std::unique_ptr<Delegate> delegate)
+    : page_(std::move(page)), prefs_(prefs), delegate_(std::move(delegate)) {
+  DCHECK(prefs_);
+  DCHECK(delegate_);
+  pref_change_registrar_.Init(prefs);
+  // Watch for external changes to containers list (e.g. sync, other windows)
+  pref_change_registrar_.Add(
+      prefs::kContainersList,
+      base::BindRepeating(&SettingsPageHandler::OnContainersChanged,
+                          base::Unretained(this)));
+}
+
+SettingsPageHandler::~SettingsPageHandler() {}
+
+void SettingsPageHandler::GetContainers(GetContainersCallback callback) {
+  std::move(callback).Run(GetContainerList(*prefs_));
```

Comment:
`Containers` vs `ContainerList` should be consistent, but also https://github.com/brave/brave-core/pull/29230/files#r2136342679